src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -1852,6 +1852,41 @@ public Mono<Void> deleteNode(UUID studyUuid, UUID nodeId, boolean deleteChildren
             .doOnError(throwable -> LOGGER.error(throwable.toString(), throwable));
     }
 
+    public Mono<Void> reindexAllStudies() {
+        List<StudyEntity> studies = studyRepository.findAll();
+        return Mono.fromRunnable(() -> studies.parallelStream().forEach(studyEntity -> reindexStudy(studyEntity.getId()).subscribe()));
+    }
+
+    public Mono<Void> reindexStudy(UUID studyUuid) {
+        Optional<StudyEntity> studyEntity = studyRepository.findById(studyUuid);
+        if (studyEntity.isPresent()) {
+            StudyEntity study = studyEntity.get();
+
+            CreatedStudyBasicInfos studyInfos = toCreatedStudyBasicInfos(study);
+            UUID networkUuid = study.getNetworkUuid();
+
+            // reindex study in elasticsearch
+            studyInfosService.recreateStudyInfos(studyInfos);
+
+            // reindex study network equipments in elasticsearch
+            String path = UriComponentsBuilder.fromPath(DELIMITER + NETWORK_CONVERSION_API_VERSION + "/networks/{networkUuid}/reindex-all")
+                .buildAndExpand(networkUuid)
+                .toUriString();
+
+            return webClient.post()
+                .uri(networkConversionServerBaseUri + path)
+                .retrieve()
+                .onStatus(httpStatus -> httpStatus != HttpStatus.OK, clientResponse -> Mono.error(new StudyException(NETWORK_INDEXATION_FAILED)))
+                .bodyToMono(Void.class)
+                .doOnError(throwable -> LOGGER.error(throwable.toString(), throwable))
+                // invalid all built model nodes in study
+                .then(invalidateBuildStatus(networkModificationTreeService.getStudyRootNodeUuid(studyUuid), false))
```

Comment:
Is it necessary to invalidate build status of root node since root node has not been modified ?

Reply:
root node has no build status, so this node won't be invalidated ...